hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -665,47 +668,63 @@ onOpenNetworkReqSn ::
   -- TODO: get rid of this (how to handle 'require' from spec?)
   Event tx ->
   Outcome tx
-onOpenNetworkReqSn env ledger st otherParty sn txs ev
-  | (number . getSnapshot) confirmedSnapshot + 1 == sn
-      && isLeader parameters otherParty sn
-      && not (snapshotPending seenSnapshot) =
-    -- TODO: Also we might be robust against multiple ReqSn for otherwise
-    -- valid request, which is currently leading to 'Error'
-    -- TODO: Verify the request is signed by (?) / comes from the leader
-    -- (Can we prove a message comes from a given peer, without signature?)
-    case applyTransactions ledger (getField @"utxo" $ getSnapshot confirmedSnapshot) txs of
-      Left (_, err) ->
-        -- FIXME: this will not happen, as we are always comparing against the
-        -- confirmed snapshot utxo?
-        Wait $ WaitOnNotApplicableTx err
-      Right u ->
-        let nextSnapshot = Snapshot sn u txs
-            snapshotSignature = sign signingKey nextSnapshot
-         in NewState
-              ( Open
-                  st
-                    { coordinatedHeadState =
-                        coordinatedHeadState
-                          { seenSnapshot = SeenSnapshot nextSnapshot mempty
-                          }
-                    }
-              )
-              [NetworkEffect $ AckSn party snapshotSignature sn]
-  | sn > (number . getSnapshot) confirmedSnapshot
-      && isLeader parameters otherParty sn =
-    -- TODO: How to handle ReqSN with sn > confirmed + 1
-    -- This code feels contrived
-    case seenSnapshot of
-      SeenSnapshot{snapshot}
-        | number snapshot == sn -> Error (InvalidEvent ev (Open st))
-        | otherwise -> Wait $ WaitOnSnapshotNumber (number snapshot)
-      _ -> Wait WaitOnSeenSnapshot
-  | otherwise = Error $ InvalidEvent ev (Open st)
+onOpenNetworkReqSn env ledger st otherParty sn txs ev =
+  requireValidReqSn $
+    waitNoSnapshotInFlight $
+      -- TODO: Also we might be robust against multiple ReqSn for otherwise
+      -- valid request, which is currently leading to 'Error'
+      -- TODO: Verify the request is signed by (?) / comes from the leader
+      -- (Can we prove a message comes from a given peer, without signature?)
+
+      -- Spec: wait U̅ ◦ T ̸= ⊥ combined with Û ← Ū̅ ◦ T
```

Comment:
Should we add some spec glossary somewhere? I don't know what these symbols mean (a part from ⊥)

Reply:
I'd say the spec _is_ the glossary (section 6.3 right now). If these comments are not helpful I can remove them.